compiler/qsc_frontend/src/typeck.rs

```diff
@@ -101,4 +101,8 @@ enum ErrorKind {
     #[diagnostic(help("replace this hole with an expression of the expected type"))]
     #[diagnostic(code("Qsc.TypeCk.TyHole"))]
     TyHole(Ty, #[label] Span),
+    #[error("insufficient type information to infer type {0}")]
+    #[diagnostic(help("provide a type annotation"))]
+    #[diagnostic(code("Qsc.TypeCk.AmbiguousType"))]
+    AmbiguousType(#[label] Span),
```

Comment:
Maybe just `Ambiguous`? (Or at least `AmbiguousTy` to be consistent with the other error names.)